### DISCUSSION TITLE
Errors and solutions

### DISCUSSION THREAD

**Discussion by span4ev** [majikat768/HuntStatsLogger#73] [Show and tell]
# --- Error:
_TypeError: unsupported operand type(s) for |: 'PyQt6.sip.wrappertype' and 'NoneType'_

# Original code (replace all such lines in 4 places, in two files)
`def __init__(self, parent: QWidget | None = None) -> None:`

# Corrected version:
`def __init__(self, parent=None) -> None:`

############################################

# --- Error:
    _match monster:
          ^
SyntaxError: invalid syntax_

# --- Original code:
```
def sanitizeMonsterNames(self, monster):
    match monster:
        case 'waterdevil': return 'Water Devil'
        case _: return monster.capitalize()      
```

# --- Corrected version:
```
def sanitizeMonsterNames(self, monster):
    if monster == 'waterdevil':
        return 'Water Devil'
    else:
        return monster.capitalize()
```

############################################

# --- Error:

  _for c in text:
TypeError: 'NoneType' object is not iterable_

# --- Original code:
```
def setText(self, text: str) -> None:
    newText = ""
    for c in text:
        if c == '\t':
            newText += ' '*tabstop
        else:
            newText += c
    return super().setText(newText)
```

# --- Corrected version:
```
def setText(self, text: str) -> None:
    if text is not None:
        newText = ""
        for c in text:
            if c == '\t':
                newText += ' ' * tabstop
            else:
                newText += c
        return super().setText(newText)
```

**Comment by majikat768** [comment]
Thanks!
I don't think I've seen any of those errors, but I can add the fixes in.
